src/FormInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+interface FormInterface
+{
+    public function addError(string $attribute, string $error): void;
```

Comment:
I don't sure that this method should be in the `FormInterface` and have `public` modifier

Reply:
This method allows addError in the validation, it must be public.